### PR TITLE
Arm resource rules

### PR DESCRIPTION
**Pull request by jianyexi**
- [x] https://github.com/Azure/azure-openapi-validator/issues/335  - LroPatch202
- [x] https://github.com/Azure/azure-openapi-validator/issues/334  - LroLocationHeader
- [x] https://github.com/Azure/azure-openapi-validator/issues/333 - XmsLongRunningOperationOptions
- [x] https://github.com/Azure/azure-openapi-validator/issues/332  - ProvisioningStateValidation
- [x] https://github.com/Azure/azure-openapi-validator/issues/331 - GetOperation200
- [x] https://github.com/Azure/azure-openapi-validator/issues/330 - DeleteResponseBodyEmpty
- [x] https://github.com/Azure/azure-openapi-validator/issues/329  - AllResourcesMustHaveDelete
- [x] https://github.com/Azure/azure-openapi-validator/issues/327 - PatchIdentityProperty
- [x] https://github.com/Azure/azure-openapi-validator/issues/326  - PatchSkuProperty
- [x] https://github.com/Azure/azure-openapi-validator/issues/324 - ConsistentPatchProperties
- [x] https://github.com/Azure/azure-openapi-validator/issues/323  - UnSupportedPatchProperties
- [x] https://github.com/Azure/azure-openapi-validator/issues/307 - TrackedResourceBeyondsThirdLevel
- [x] https://github.com/Azure/azure-openapi-validator/issues/306 - TrackedResourcesMustHavePut

### CODE REVIEW

**Review by mikekistler — APPROVED** [review]
This looks good, but on my system two tests fail:
    ✕ no delete in for tracked resource (11 ms)
    ✕ tracked resource beyonds third level (2 ms)

I haven't figured out how to debug rules in this repo yet, so I can't say why these fail -- sorry.

And I see that we don't actually run the tests in the PR checks -- shouldn't we?

I left a couple other minor comments/suggestions.

Finally, I think other developers and "future you" would appreciate a few more comments in the code, particularly function heaaders that describe the intended behavior of the function.

I'm sure you can work these out so I will approve.

File: packages/rulesets/src/spectral/functions/validate-patch-body-param-properties.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { getGetOperationSchema, getProperties } from "./utils"
```
[the remark this answers is not in the document]
Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/validate-patch-body-param-properties.ts:1]: Oh, right, thanks for reminding me

File: packages/rulesets/src/spectral/functions/lro-patch-202.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export const lroPatch202 = (patchOp: any, _opts: any, ctx: any) => {
+  if (patchOp === null || typeof patchOp !== "object") {
+    return []
+  }
+  const path = ctx.path || []
+  if (!patchOp["x-ms-long-running-operation"]) {
+    return []
+  }
+  const errors: any = []
+  if (patchOp?.responses && !patchOp?.responses["202"]) {
```
[the remark this answers is not in the document]
Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/lro-patch-202.ts:10]: will remove it

**Comment by jianyexi** [comment]
@mikekistler how do you test js ruleset ? one way I usually use is you can auto attach the node process see https://code.visualstudio.com/blogs/2018/07/12/introducing-logpoints-and-auto-attach#_autoattaching-to-node-processes , this works for 'jest' as well

**Pull request merged**